### ISSUE TITLE
Design the generation journey for GPT2

### ISSUE THREAD

**Issue by chenmoneygithub** [keras-team/keras-nlp#447]
Unsure if it should be a head or a class method, let's figure it out!

**Comment by jbischof** [comment]
Relevant reading: https://huggingface.co/blog/tf-xla-generate

Also want to incorporate our TBD preprocessing redesign

**Comment by mattdangerw** [comment]
Also see the discussion here https://github.com/keras-team/keras-nlp/issues/415

Potentially this could be both a task model (like BertClassifier) AND a collection of methods on the class. E.g. if we want our backbones to always return a dense representation and not language model logits, then you might want something like `GPT2TextGenerator` which adds the language model logits outputs, and then add some sort of generation utilities directly to that class `GPT2TextGenerator.generate()` or `GPT2TextGenerator.beam_search()` or whatever we end up calling it.

**Comment by chenmoneygithub** [comment]
One tough point is how to reuse the generation code, I don't like the mixin solution because it makes the code go wild. It appears using a task model + our text generation util could work, but I need to take a closer look onto that.

**Comment by mattdangerw** [comment]
Fixed
